packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -14,6 +14,13 @@ function reducer(state, action) {
     case actionTypes.RESET_STATE:
       return action.initialState
     case actionTypes.LOADING:
+      // if previous action resulted in error - refetch will reset the state
+      if (state.error) {
+        return {
+          ...action.initialState,
```

Comment:
@edvinasbartkus this will clear the `state.data` which could be set from a previous call - we should only clear the error and set loading to true.

Reply:
but can we have a state where data is available and also error would be true?

Maybe the best would be only to persist `data`, something like that:
```js
return {
  ...action.initialState,
  data: state.data,
  loading: true
}
```

otherwise, I guess we need to do something like that to cleanup all error properties?
```js
return {
  ...state,
  error: false,
  fetchError: undefined,
  httpError: undefined,
  graphQLErrors: undefined
}
```

Reply:
@edvinasbartkus this scenario can happen when calling `refetch`... Yea I'd probably go with the first option 👍